### PR TITLE
Migrate chef, puppet, salt, and ansible deployment test jobs from CircleCI to GitHub

### PR DESCRIPTION
**Pull request by nitaliya**
Migrate `chef`, `puppet`, `salt`, and `ansible` deployment test jobs 
Ticket: [OTL-1324](https://signalfuse.atlassian.net/browse/OTL-1324). 

Changes:
- Adding GitHub Action jobs of `chef`, `puppet`, `salt`, and `ansible` deployment test. 
- Removing CircleCi `deployment_test` jobs.  

Note:
- Instead of using `git diff` in the script, pushed changes uses https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestpaths to find the relevant changes related to the deployment type.
- Advantage of using the `onpushpull_requestpaths` is:
  - we let GitHub find the code difference.
  - in the future, if we have to make the change specific to a deployment pipeline (ex: only puppet), we won't have to worry about other deployments.

### CODE REVIEW

File: .github/workflows/ansible_deployment_test.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      - 'tests/packaging/common.py'
+      - '.github/scripts/run-pytest.sh'
+
+concurrency:
+  group: ansible_deployment_test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  RESULT_PATH: testresults
+  PYTHON_VERSION: 3.8.0
```
[the remark this answers is not in the document]
Comment (nitaliya) [reply .github/workflows/ansible_deployment_test.yml]: I have tried with python version 3.8.10. 
Could You Please Clarify, why we have use this specific version of python?

File: .github/workflows/ansible_deployment_test.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  REQUIREMENTS_PATH: "tests/requirements.txt"
+  GO_VERSION: 1.14.2
```
[the remark this answers is not in the document]
Comment (nitaliya) [reply .github/workflows/ansible_deployment_test.yml]: Changes made.
Thanks for the correction!

File: .github/workflows/ansible_deployment_test.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/pip
+          key: v1-pytest-${{ env.PYTHON_VERSION }}-${{ hashFiles(env.REQUIREMENTS_PATH) }}
+
+      - name: Install pytest
+        run: |
+          if which pip; then
```
[the remark this answers is not in the document]
Comment (nitaliya) [reply .github/workflows/ansible_deployment_test.yml]: I have tried removing the downgrading step of PIP and using the latest version of PIP it is working fine, so made this change.
From this now my understanding is in CircleCI we are using an older version of PIP but now while migrating, we are upgrading the version of PIP. Correct me if I am wrong.

File: .github/workflows/ansible_deployment_test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcheng-splunk) [reply .github/workflows/ansible_deployment_test.yml]: We recently upgraded the bundled python to 3.8.10 (https://github.com/signalfx/signalfx-agent/blob/main/Dockerfile#L82), so I think we should use the same version for tests.

File: .github/workflows/ansible_deployment_test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jcheng-splunk) [reply .github/workflows/ansible_deployment_test.yml]: Correct.  As long as there are no issues, we can just use the provided pip version.

**Review by jcheng-splunk — APPROVED** [review]

**Pull request merged**